### PR TITLE
Add ReconcileVirtualService function to pkg

### PR DESCRIPTION
**Pull request by ZhiminXiang**
The `ReconcileVirtualService` is copied from https://github.com/knative/serving/blob/9599eb48d62f4ac0884b046969e5bc85760fb41b/pkg/reconciler/ingress/ingress.go#L356.

And I will use `ReconcileVirtualService` to replace it once we check in this PR.

### CODE REVIEW

File: reconciler/resources/istio/virtualservice.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package istio
```
Comment (mattmoor) [inline reconciler/resources/istio/virtualservice.go]: missing license

File: reconciler/resources/istio/virtualservice_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
+package istio
```
Comment (mattmoor) [inline reconciler/resources/istio/virtualservice_test.go]: license

File: reconciler/resources/istio/virtualservice_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		Namespace: "default",
+		UID:       "abcd",
+	},
+}
+var isControl = true
+var ownerRef = metav1.OwnerReference{
+	Kind:       ownerObj.Kind,
+	Name:       ownerObj.Name,
+	UID:        ownerObj.UID,
+	Controller: &isControl,
```
Comment (mattmoor) [inline reconciler/resources/istio/virtualservice_test.go]: `ptr.Bool(true)`

File: reconciler/resources/istio/virtualservice_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		Name:            "vs",
+		Namespace:       "default",
+		OwnerReferences: []metav1.OwnerReference{ownerRef},
+	},
+	Spec: v1alpha3.VirtualServiceSpec{
+		Hosts: []string{"origin.example.com"},
+	},
+}
+
+var desired = &v1alpha3.VirtualService{
```
Comment (mattmoor) [inline reconciler/resources/istio/virtualservice_test.go]: use a `var` block:
```
var (
...
)
```

File: reconciler/resources/istio/virtualservice.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+// VirtualServiceAccessor is an accessor to get/write VirtualService.
+type VirtualServiceAccessor struct {
+	SharedClientSet      sharedclientset.Interface
+	VirtualServiceLister istiolisters.VirtualServiceLister
+}
+
+// ReconcileVirtualService reconciles VirtualService to the desired states.
+func (v *VirtualServiceAccessor) ReconcileVirtualService(ctx context.Context, owner kmeta.Accessor, desired *v1alpha3.VirtualService, recorder record.EventRecorder) (*v1alpha3.VirtualService, error) {
```
Comment (mattmoor) [inline reconciler/resources/istio/virtualservice.go]: I don't love this signature....

**Comment by ZhiminXiang** [comment]
Per our offline discussion, we need a new pattern for implementing this PR and it would be better to put this into knative/serving first.

**Pull request closed without merging**